msal/src/main/java/com/microsoft/identity/client/internal/controllers/LocalMSALController.java

```diff
@@ -265,4 +273,39 @@ public AcquireTokenResult acquireTokenSilent(
 
         return acquireTokenSilentResult;
     }
+
+    public void removeLocalAccount(@Nullable final IAccount account,
+                                   @NonNull final PublicClientApplicationConfiguration configuration,
+                                   @NonNull final PublicClientApplication.AccountsRemovedCallback callback) {
+        // FEATURE SWITCH: Set to false to allow deleting Accounts in a tenant-specific way.
+        final boolean deleteAccountsInAllTenants = true;
+
+        final String realm = deleteAccountsInAllTenants ? null : AccountAdapter.getRealm(account);
+
+        // Step 1. clear msal local cache
+        final boolean localRemoveAccountSuccess = !configuration
+                .getOAuth2TokenCache()
+                .removeAccount(
+                        account.getEnvironment(),
+                        configuration.getClientId(),
+                        account.getHomeAccountIdentifier().getIdentifier(),
+                        realm
+                ).isEmpty();
+
+        // Step 2.
+        // Clear cookies from embedded webView.
+        final CookieManager cookieManager = CookieManager.getInstance();
```

Comment:
Given that I also have this code change in Broker (FLW global sign out), should we extract this function to common?

Reply:
FLW is a different scenario. FLW is restricted with one account with a device level scenario. Whereas, the normal local MSAL follow is mutli-accounts with a application level supported. I am not inclined to combine the removeAccounts() together.

Reply:
oh no. I'm referring to this 'clear cookie' functionality specifically.

Reply:
Could you kindly link the code here?

Reply:
@heidijinxujia @rpdome's work is here https://github.com/AzureAD/ad-accounts-for-android/pull/951/files

As Dome calls out, looks like we need a shared function here

Reply:
I moved the code to common https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/491